Add document:deleteFields (#1928)
kuzzleio/kuzzle: doc/2/api/controllers/document/delete-fields/index.md

---
code: true
type: page

> title: delete-fields

Reviewer: Page title must match method name
```suggestion
title: deleteFields
```

description: Delete fields of an existing document.
---

# deleteFields

Delete fields of an existing document.

---

## Query Syntax

### HTTP

```http
URL: http://kuzzle:7512/<index>/<collection>/<documentId>/_fields[?refresh=wait_for]
Method: DELETE
Body:
```

```js
{
  "fields": [
    // path of fields to remove
  ]
}
```

### Other protocols

```js
{
  "index": "<index>",
  "collection": "<collection>",
  "controller": "document",
  "action": "deleteFields",
  "_id": "<documentId>",
  "refresh": "wait_for",
  "body": {
    "fields": [
      // path of fields to remove
    ]
  }
}
```

---

## Arguments

- `collection`: collection name
- `documentId`: unique ID of the document where the fields should be removed
- `index`: index name

### Optional:

- `refresh`: if set to `wait_for`, Kuzzle will not respond until the new document content is indexed

---

## Body properties

> - `fields`: an array of string. Each string represent a path (see lodash path) to a specific field to remove

Reviewer 2: ```suggestion
- `fields`: an array of string. Each string represents a path (see [lodash path](https://lodash.com/docs/4.17.15#toPath)) to a specific field to remove
```

> - `fields`: an array of string. Each string represents a path (see [lodash path](https://lodash.com/docs/4.17.15#toPath)) to a specific field to remove

Reviewer 3: ```suggestion
- `fields`: an array of strings. Each string represents a path (see [lodash path](https://lodash.com/docs/4.17.15#toPath)) to a specific field to remove
```

---

## Response

Returns an object containing updated document information, with the following properties:

- `_id`: document unique identifier

> - `_source`: new document content

Reviewer: Maybe we could add a `source` option to this method like for the `document:update` method so Kuzzle will return the body only if asked.

What do you think?

Author: This could reduce the network load is the user doesn't need to get the new document, I will add an option for that

---
code: true
type: page

> title: delete-fields

Reviewer: Missing `description`

---

# deleteFields

Delete fields of an existing document.

---

## Query Syntax

### HTTP

```http
URL: http://kuzzle:7512/<index>/<collection>/<documentId>/_fields[?refresh=wait_for]
Method: DELETE

> Body: {}

Reviewer: Nitpicking
```suggestion
Body: 
```

```

```js
{
  "fields": [
    // path of fields to remove
  ]
}
```

### Other protocols

```js
{
  "index": "<index>",
  "collection": "<collection>",
  "controller": "document",

>   "action": "replace",

Reviewer: ```suggestion
  "action": "deleteFields",
```

  "_id": "<documentId>",
  "refresh": "wait_for",
  "body": {
    "fields": [
      // path of fields to remove
    ]
  }
}
```

---

## Arguments

- `collection`: collection name
- `documentId`: unique ID of the document to replace
- `index`: index name

### Optional:

- `refresh`: if set to `wait_for`, Kuzzle will not respond until the new document content is indexed

---

## Body properties

- `fields`: an array of string. Each string represent a path (see lodash path) to a specific field to remove

---

## Response

Returns an object containing updated document information, with the following properties:

- `_id`: document unique identifier
- `_source`: new document content
- `_version`: updated document version number

```js
{
  "status": 200,
  "error": null,
  "index": "<index>",
  "collection": "<collection>",
  "controller": "document",

>   "action": "replace",

Reviewer: ```suggestion
  "action": "deleteFields",
```
